### PR TITLE
Reset online clients when stream is disconnected

### PR DESCRIPTION
**Pull request by 7hong13**
#### What are the relevant tickets?

Fixes #
- https://github.com/yorkie-team/yorkie-js-sdk/pull/796

### Checklist

- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: yorkie/src/main/kotlin/dev/yorkie/core/Presences.kt [hunk 1]
```diff
@@ -18,9 +18,9 @@ public class Presences private constructor(
 
     public operator fun minus(actorID: ActorID): Presences = (map - actorID).asPresences()
 
-    override fun hashCode(): Int = map.hashCode()
-
-    override fun equals(other: Any?): Boolean = map == (other as? Presences)?.map
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/core/Presences.kt:23]: I removed it to ensure that `document.publishPresenceEvent()` is properly invoked whenever a new value is emitted to `document.presences`.

File: yorkie/src/main/kotlin/dev/yorkie/core/Client.kt [hunk 2]
```diff
@@ -290,14 +291,14 @@ public class Client @VisibleForTesting internal constructor(
...
                     latestStream = it
+                    attachment.document.publishEvent(StreamConnectionChanged.Connected)
                 }
                 val streamJob = launch(start = CoroutineStart.UNDISPATCHED) {
                     val channel = stream.responseChannel()
                     var retry = 0
                     while (!stream.isReceiveClosed() && !channel.isClosedForReceive) {
                         val receiveResult = channel.receiveCatching()
                         receiveResult.onSuccess {
-                            attachment.document.publishEvent(StreamConnectionChange.Connected)
```
Comment (skhugh) [inline yorkie/src/main/kotlin/dev/yorkie/core/Client.kt:300]: don't we need this line?
if the stream fails then succeed on retry, users would not receive the connected state.

Comment (7hong13) [reply yorkie/src/main/kotlin/dev/yorkie/core/Client.kt:300]: I missed it. fixed in [4f46596](https://github.com/yorkie-team/yorkie-android-sdk/pull/181/commits/4f465969d8507f4916feeaeec7e84af1603792aa)

**Review by skhugh — APPROVED** [review]

**Comment by 7hong13** [comment]
Some tests are failing in the CI, but I checked they're working fine with the real server. I'll investigate these failing tests after merging.

**Pull request merged**